### ISSUE TITLE
Pass `ownProps` to `authSelector`

### ISSUE THREAD

**Issue by themouette** [mjrussell/redux-auth-wrapper#20]
Redux passes component properties to `mapStateToProps`, What do you think about passing them to `authSelector` as well?

It allow to leverage other Higher Order Components, accept configuration or access route params.

What do you think?

**Comment by themouette** [comment]
To precise what I have in mind:

``` javascript
// Redirects to /login by default
const UserIsAuthenticated = UserAuthWrapper({
  authSelector: (state, ownProps) => state.user, // how to get the user state
  redirectAction: routeActions.replace, // the redux action to dispatch for redirect
  wrapperDisplayName: 'UserIsAuthenticated' // a nice name for this auth check
})
```

**Comment by mjrussell** [comment]
Yeah makes perfect sense. Going to work this and the onEnter and release something today.

**Comment by themouette** [comment]
You're awesome!
Le 15 févr. 2016 18:22, "Matthew Russell" <notifications@github.com> a
écrit :

**Comment by mjrussell** [comment]
> You're awesome!
:+1: :smile: 

One caveat, if you are using the `onEnter` method you will not be able to use the `ownProps`. Im thinking about passing a third argument which is `isOnEnter` which is a bool to check against.

**Comment by mjrussell** [comment]
Just released v0.3.0 with `ownProps` support
